api/src/main/java/org/openmrs/module/commonreports/reports/DRCARTDeathReportManager.java

```diff
@@ -0,0 +1,264 @@
+	}
+	
+	private Parameter getEndDateParameter() {
+		return new Parameter("endDate", MessageUtil.translate("commonreports.report.util.reportingEndDate"), Date.class);
+	}
+	
+	public static String col1 = "";
+	
+	public static String col2 = "";
+	
+	public static String col3 = "";
+	
+	public static String col4 = "";
+	
+	public static String col5 = "";
+	
+	public static String col6 = "";
+	
+	public static String col7 = "";
+	
+	public static String col8 = "";
+	
+	public static String col9 = "";
+	
+	public static String col10 = "";
+	
+	public static String col11 = "";
+	
+	@Override
+	public List<Parameter> getParameters() {
+		List<Parameter> params = new ArrayList<Parameter>();
+		params.add(getStartDateParameter());
+		params.add(getEndDateParameter());
+		return params;
+	}
+	
+	@Override
+	public ReportDefinition constructReportDefinition() {
+		ReportDefinition rd = new ReportDefinition();
+		rd.setUuid(getUuid());
+		rd.setName(getName());
+		rd.setDescription(getDescription());
+		rd.setParameters(getParameters());
+		
+		// artDeath Grouping
+		CohortCrossTabDataSetDefinition artDeath = new CohortCrossTabDataSetDefinition();
+		artDeath.addParameters(getParameters());
+		rd.addDataSetDefinition(getName(), Mapped.mapStraightThrough(artDeath));
+		
+		Map<String, Object> parameterMappings = new HashMap<String, Object>();
+		parameterMappings.put("onOrAfter", "${startDate}");
+		parameterMappings.put("onOrBefore", "${endDate}");
+		SqlCohortDefinition sqd = new SqlCohortDefinition();
+		
+		// Death Date in range
+		// ART Start/Initiation before death date
+		String sql = "SELECT DISTINCT p.patient_id FROM patient p WHERE p.voided = 0 "
+		        + "AND EXISTS (SELECT 1 FROM obs o_date JOIN concept c_date ON o_date.concept_id = c_date.concept_id WHERE o_date.person_id = p.patient_id AND c_date.uuid = '1543AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA' AND o_date.voided = 0 AND o_date.value_datetime BETWEEN :onOrAfter AND :onOrBefore AND o_date.obs_datetime = (SELECT MAX(od_inner.obs_datetime) FROM obs od_inner WHERE od_inner.person_id = p.patient_id AND od_inner.concept_id = c_date.concept_id AND od_inner.voided = 0)) "
+		        + "AND EXISTS (SELECT 1 FROM obs o_earlier JOIN concept c_earlier ON o_earlier.concept_id = c_earlier.concept_id WHERE o_earlier.person_id = p.patient_id AND c_earlier.uuid = '159599AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA' AND o_earlier.voided = 0 AND o_earlier.value_datetime < (SELECT latest_date_obs.value_datetime FROM obs latest_date_obs JOIN concept latest_date_concept ON latest_date_obs.concept_id = latest_date_concept.concept_id WHERE latest_date_obs.person_id = p.patient_id AND latest_date_concept.uuid = '1543AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA' AND latest_date_obs.voided = 0 AND latest_date_obs.obs_datetime = (SELECT MAX(ld_inner.obs_datetime) FROM obs ld_inner WHERE ld_inner.person_id = p.patient_id AND ld_inner.concept_id = latest_date_concept.concept_id AND ld_inner.voided = 0)));";
```

Comment:
Looking at this, I guess I've missed this before, but we're really recording patient death as an obs instead of via updating the patient?

Reply:
@ibacher I was depending on the patient being updated as dead. This is a unique case where we need this form filled specifically because the death is HIV related. See [here](https://github.com/path-drc/openmrs-module-commonreports/pull/3/files#diff-66290b54de4d9e9c5c2277f94b95984422665272e1df2113765ffa5abdd70f65R144-R145) for how am handling dead/live patients
It makes sense to add this logic here too. Let me push that as well